pyxem/generators/pdf_generator.py

```diff
@@ -0,0 +1,96 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017-2019 The pyXem developers
+#
+# This file is part of pyXem.
+#
+# pyXem is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# pyXem is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with pyXem.  If not, see <http://www.gnu.org/licenses/>.
+
+"""PDF generator and associated tools.
+
+"""
+
+import numpy as np
+
+from hyperspy.signals import Signal1D
+
+from pyxem.signals.reduced_intensity1d import ReducedIntensity1D
+from pyxem.signals.pair_distribution_function1d import PairDistributionFunction1D
+from pyxem.signals import transfer_navigation_axes
+
+
+class PDFGenerator():
+    """Generates a PairDistributionFunction1D signal from a specified
+        ReducedIntensity1D signal.
+
+
+    Parameters
+    ----------
+    signal : ReducedIntensity1D
+        A reduced intensity radial profile.
+    """
+
+    def __init__(self, signal, *args, **kwargs):
+        self.signal = signal
+
+    def get_pdf(self,
+                s_cutoff,
+                r_cutoff=[0, 20],
+                r_increment=0.01
+                ):
+        """ Calculates the pdf from the reduced intensity signal.
+
+        Parameters
+        ----------
+        s_cutoff : list of float
+                    A list with the format [<s_min>, <s_max>], which sets the
+                    integration limits for the pdf calculation. Note that s is
+                    defined here as s = 2 sin(theta)/lambda = 1/d.
+        r_cutoff : list of float
+                    A list with the format [<r_min>, <r_max>], which sets the
+                    limits of the real space axis in the calculated PDF.
+        r_increment : float
+                    Step size in r in the extracted PDF.
+        """
+        r_min, r_max = r_cutoff[0],r_cutoff[1]
+        s_min, s_max = s_cutoff[0],s_cutoff[1]
+
+        r_values = np.arange(r_min, r_max, r_increment)
+        r_values = r_values.reshape(1, r_values.size)
+
+        s_scale = self.signal.axes_manager.signal_axes[0].scale
+        s_limits = [int(s_min / s_scale), int(s_max / s_scale)]
+
+        #check that these aren't out of bounds
+        if s_limits[1] > self.signal.axes_manager.signal_axes[0].size:
```

Comment:
I don't like this very much. I think if you set s_max too large the response should be a ValueError so that the user has to specify it correctly. Automatically "fixing" the problem like this might cause users to not fully understand that they're doing something wrong and why.